src/Adapters/Auth/apple.js

```diff
@@ -2,34 +2,48 @@
 // https://developer.apple.com/documentation/signinwithapplerestapi
 
 const Parse = require('parse/node').Parse;
-const httpsRequest = require('./httpsRequest');
-const NodeRSA = require('node-rsa');
+const jwksClient = require('jwks-rsa');
+const util = require('util');
 const jwt = require('jsonwebtoken');
 
 const TOKEN_ISSUER = 'https://appleid.apple.com';
 
 let currentKey;
 
-const getApplePublicKey = async () => {
-  let data;
-  try {
-    data = await httpsRequest.get('https://appleid.apple.com/auth/keys');
-  } catch (e) {
-    if (currentKey) {
-      return currentKey;
-    }
-    throw e;
+const getAppleKeyByKeyId = async keyId => {
+  if (currentKey && currentKey.kid === keyId) {
```

Comment:
I believe this is handled for you in the jwksClient - cache:true and can be removed